src/Spatial/Euclidean/StraightLine2D.cs

```diff
@@ -0,0 +1,315 @@
+        /// </summary>
+        /// <param name="points">The points though which the line should pass</param>
+        /// <exception cref="ArgumentException">Thrown when less than 2 points are provided</exception>
+        /// <exception cref="ArgumentException">Thrown when points are not collinear</exception>
+        public StraightLine2D(Point2D[] points)
+        {
+            if (points.Length < 2)
+            {
+                throw new ArgumentException("A line must be constructed from at least two points");
+            }
+
+            Gradient = (points[1].Y - points[0].Y) / (points[1].X - points[0].X);
+            if (double.IsInfinity(Gradient))
+            {
+                yoffset = double.NaN; // special case vertical line has no y-intercept
+                xoffset = points[1].X;
+            }
+            else
+            {
+                yoffset = points[1].Y - (Gradient * points[1].X);
+                xoffset = double.NaN;
+            }
+
+            for (int i = 2; i < points.Length; i++)
+            {
+                if (!IsThrough(points[i]))
+                {
+                    throw new ArgumentException("The points provided are not collinear");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Determines if the line is horizontal
+        /// </summary>
+        /// <returns>True if the line is horizontal</returns>
+        public bool IsHorizontal()
+        {
+            return Gradient == 0;
+        }
+
+        /// <summary>
+        /// Determines if the line is vertical
+        /// </summary>
+        /// <returns>returns true if the line is vertical</returns>
+        public bool IsVertical()
+        {
+            return double.IsInfinity(Gradient);
+        }
+
+        /// <summary>
+        /// Tests if the other line is parallel to this one
+        /// </summary>
+        /// <returns>Returns true if lines are parallel</returns>
+        public bool IsParallel(ILine otherLine)
+        {
+            if (otherLine is StraightLine2D)
+            {
+                IsParallel((StraightLine2D)otherLine);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Tests if the other line is parallel to this one
+        /// </summary>
+        /// <returns>Returns true if lines are parallel</returns>
+        public bool IsParallel(StraightLine2D otherLine)
+        {
+            return otherLine.Gradient == Gradient;
+        }
+
+        /// <summary>
+        /// Finds the midpoint of two points along the current line
+        /// </summary>
+        /// <param name="pointA">a point on the line</param>
+        /// <param name="pointB">a point on the line</param>
+        /// <returns>A point midway between the two provided points</returns>
+        public Point2D MidPoint(Point2D pointA, Point2D pointB)
+        {
+            return new Point2D((pointA.X + pointB.X) / 2, (pointA.Y + pointB.Y) / 2);
```

Comment:
How does this find the midpoint 'along the current line'? Also there is `Point2D.MidPoint()` IIRC.

Reply:
midpoint formula - http://www.purplemath.com/modules/midpoint.htm
it so happens that the line is straight - if it was curved this would be different - and yes I know its not part of the ILine2D interface yet, it will be eventually.

Reply:
But there is nothing saying the midpoint is on the line.

Reply:
good point - will fix